shared/lib/trace.test.ts

```diff
@@ -0,0 +1,112 @@
+import { Span, startSpan, withIsolationScope } from '@sentry/browser';
+import { trace } from './trace';
+
+jest.mock('@sentry/browser', () => ({
+  withIsolationScope: jest.fn(),
+  startSpan: jest.fn(),
+}));
+
+const NAME_MOCK = 'testTransaction';
+const PARENT_CONTEXT_MOCK = {} as Span;
+
+const TAGS_MOCK = {
+  tag1: 'value1',
+  tag2: true,
+  tag3: 123,
+};
+
+const DATA_MOCK = {
+  data1: 'value1',
+  data2: true,
+  data3: 123,
+};
+
+function mockGetMetaMetricsEnabled(enabled: boolean) {
+  global.sentry = {
+    getMetaMetricsEnabled: () => Promise.resolve(enabled),
+  };
+}
+
+describe('Trace', () => {
+  const startSpanMock = jest.mocked(startSpan);
+  const withIsolationScopeMock = jest.mocked(withIsolationScope);
+  const setTagsMock = jest.fn();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    startSpanMock.mockImplementation((_, fn) => fn({} as Span));
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    withIsolationScopeMock.mockImplementation((fn: any) =>
+      fn({ setTags: setTagsMock }),
+    );
+  });
+
+  describe('trace', () => {
+    // @ts-expect-error This function is missing from the Mocha type definitions
```

Comment:
I believe this API does not exist in Mocha, only in Jest. If this test runs in Jest, is there a way to reference the type explicitly to avoid the error (with the `as` keyword)? Alternatively we could wrap `it` in a `forEach`.

Reply:
This comment is a standard in a few other places, so I don't doubt it can be resolved with better types or updating config files, but perhaps best in an alternate PR to fix them all?